Source/WebCore/dom/Document.cpp

```diff
@@ -4485,6 +4485,8 @@ void Document::setRequiresTrustedTypes(bool required)
         return;
 
     frame->checkedScript()->setRequiresTrustedTypes(required);
+    if (scriptExecutionContext()->settingsValues().trustedTypesEnabled)
```

Comment:
Can you check `settings().trustedTypesEnabled()` instead?

Reply:
Ah yeah good point. Will change this and Document write to use that instead.